test/unit/paramFunctionTest.js

```diff
@@ -0,0 +1,140 @@
+/* eslint-env mocha */
+
+const assert = require('assert')
+const path = require('path')
+const generateTestApp = require('../util/generateTestApp')
+const cleanupTestApp = require('../util/cleanupTestApp')
+const fork = require('child_process').fork
+const fse = require('fs-extra')
+const request = require('supertest')
+
+describe('parameter Function Test Section', function () {
+  // path to the app Directory
+  const appDir = path.join(__dirname, '../', 'app', 'paramFunctionTest')
+
+  // specify the options that will be passed to the generateTestApp
+  let options = {rooseveltPath: '../../../roosevelt', method: 'startServer'}
+
+  beforeEach(function (done) {
+    // start by copying the alreadly made mvc directory into the app directory
+    fse.copySync(path.join(__dirname, '../', 'util', 'mvc'), path.join(appDir, 'mvc'))
+    done()
+  })
+
+  // delete the test app Directory and start with a clean state after each test
+  afterEach(function (done) {
+    cleanupTestApp(appDir, (err) => {
+      if (err) {
+        throw err
+      } else {
+        done()
+      }
+    })
+  })
+
+  it('should execute what is in onReqStart', function (done) {
+    // generate the app
+    generateTestApp({
+      appDir: appDir,
+      generateFolderStructure: true,
+      onReqStart: true,
```

Comment:
I understand why you're using `true` for these functions and then overriding them in the app.js generator, but I'm concerned that we're piling on technical debt at this point. It would be ideal to be able to custom tailor these functions in tests, and if we ever need to then we'd have to go back to all of these old tests and change them later on. We should find a way to pass functions here that get written to the file directly without overrides.